pyabc/smc.py

```diff
@@ -927,7 +923,14 @@ def run(self,
             elif acceptance_rate < min_acceptance_rate:
                 logger.info("Stopping: minimum acceptance rate.")
                 break
-
+            elif t==t_max:
+                break
+                
+            # prepare next iteration
+            self._prepare_next_iteration(
```

Comment:
I think it makes sense to move the preparation of the next iteration here. good!